Build failure: pkgsCross.mingwW64.libxcb

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

-

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
make  all-am
make[2]: Entering directory '/build/libxcb-1.17.0/src'
  CC       xcb_conn.lo
  CC       xcb_out.lo
  CC       xcb_in.lo
  CC       xcb_ext.lo
  CC       xcb_xid.lo
  CC       xcb_list.lo
  CC       xcb_util.lo
xcb_util.c: In function '_xcb_open':
xcb_util.c:258:9: warning: unused variable 'actual_filelen' [-Wunused-variable]
  258 |     int actual_filelen;
      |         ^~~~~~~~~~~~~~
xcb_util.c:257:11: warning: unused variable 'file' [-Wunused-variable]
  257 |     char *file = NULL;
      |           ^~~~
xcb_util.c:256:12: warning: unused variable 'filelen' [-Wunused-variable]
  256 |     size_t filelen;
      |            ^~~~~~~
xcb_util.c:255:17: warning: unused variable 'base' [-Wunused-variable]
  255 |     const char *base = unix_base;
      |                 ^~~~
xcb_util.c:249:9: warning: unused variable 'fd' [-Wunused-variable]
  249 |     int fd;
      |         ^~
xcb_util.c: In function '_xcb_do_connect':
xcb_util.c:362:46: error: passing argument 4 of 'setsockopt' from incompatible pointer type [-Wincompatible-pointer-types]
  362 |     setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, &on, sizeof(on));
      |                                              ^~~
      |                                              |
      |                                              int *
In file included from xcb_windefs.h:34,
                 from xcb_util.c:42:
/nix/store/1pkim64f6fqqnyqsrz8s8mhx4gck6jp0-x86_64-w64-mingw32-gcc-14.3.0/x86_64-w64-mingw32/sys-include/winsock2.h:1035:88: note: expected 'const char *' but argument is of type 'int *'
 1035 |   WINSOCK_API_LINKAGE int WSAAPI setsockopt(SOCKET s,int level,int optname,const char *optval,int optlen);
      |                                                                            ~~~~~~~~~~~~^~~~~~
xcb_util.c:363:46: error: passing argument 4 of 'setsockopt' from incompatible pointer type [-Wincompatible-pointer-types]
  363 |     setsockopt(fd, SOL_SOCKET, SO_KEEPALIVE, &on, sizeof(on));
      |                                              ^~~
      |                                              |
      |                                              int *
/nix/store/1pkim64f6fqqnyqsrz8s8mhx4gck6jp0-x86_64-w64-mingw32-gcc-14.3.0/x86_64-w64-mingw32/sys-include/winsock2.h:1035:88: note: expected 'const char *' but argument is of type 'int *'
 1035 |   WINSOCK_API_LINKAGE int WSAAPI setsockopt(SOCKET s,int level,int optname,const char *optval,int optlen);
      |                                                                            ~~~~~~~~~~~~^~~~~~
xcb_util.c: In function 'xcb_connect_to_display_with_auth_info':
xcb_util.c:541:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  541 |     WSADATA wsaData;
      |     ^~~~~~~
make[2]: *** [Makefile:1055: xcb_util.lo] Error 1
make[2]: Leaving directory '/build/libxcb-1.17.0/src'
make[1]: *** [Makefile:813: all] Error 2
make[1]: Leaving directory '/build/libxcb-1.17.0/src'
make: *** [Makefile:799: all-recursive] Error 1
```

### Additional context

_No response_

### System metadata

x86_64-linux

### Notify maintainers

no maintainers
previous non-treewides from @quantenzitrone 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/li/libxcb/package.nix
+++ b/pkgs/by-name/li/libxcb/package.nix
@@ -57,6 +57,10 @@ stdenv.mkDerivation (finalAttrs: {
     tests.pkg-config = testers.testMetaPkgConfig finalAttrs.finalPackage;
   };
 
+  env = lib.optionalAttrs stdenv.hostPlatform.isMinGW {
+    NIX_CFLAGS_COMPILE = toString [ "-Wno-incompatible-pointer-types" ];
+  };
+
   meta = {
     description = "C interface to the X Window System protocol";
     homepage = "https://gitlab.freedesktop.org/xorg/lib/libxcb";

